### ISSUE TITLE
Comparing Branches, GED Encode Error

### ISSUE THREAD

**Issue by iarepandemonium** [go-gitea/gitea#14420]
- Gitea version (or commit ref):
![image](https://user-images.githubusercontent.com/8331173/105441765-3a89d400-5c37-11eb-91c1-4005b345aa1a.png)

- Git version:
- Operating system:

- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Did not attempt
- Log gist:
```
2021/01/22 01:43:24 routers/repo/http.go:105:HTTP() [E] Attempted access of unknown user from 127.0.0.1
2021/01/22 01:43:24 routers/repo/http.go:106:HTTP() [E] GetUserByName: user does not exist [uid: 0, name: api, keyid: 0]
2021/01/22 01:43:25 routers/repo/http.go:105:HTTP() [E] Attempted access of unknown user from 127.0.0.1
2021/01/22 01:43:25 routers/repo/http.go:106:HTTP() [E] GetUserByName: user does not exist [uid: 0, name: api, keyid: 0]
2021/01/22 03:16:12 .../templates/helper.go:806:ActionContent2Commits() [E] json.Unmarshal:

        ERROR: unexpected end of JSON input
2021/01/22 03:16:22 ...ters/repo/compare.go:449:PrepareCompareDiff() [E] GetDiffRange: encoding/hex: invalid byte: U+0074 't'
2021/01/22 03:16:26 ...ters/repo/compare.go:449:PrepareCompareDiff() [E] GetDiffRange: encoding/hex: invalid byte: U+0074 't'
2021/01/22 03:16:40 ...ters/repo/compare.go:449:PrepareCompareDiff() [E] GetDiffRange: encoding/hex: invalid byte: U+0074 't'
2021/01/22 03:17:28 .../templates/helper.go:806:ActionContent2Commits() [E] json.Unmarshal:

        ERROR: unexpected end of JSON input
2021/01/22 03:17:36 ...ters/repo/compare.go:449:PrepareCompareDiff() [E] GetDiffRange: encoding/hex: invalid byte: U+0074 't'
2021/01/22 03:22:41 ...ters/repo/compare.go:449:PrepareCompareDiff() [E] GetDiffRange: encoding/hex: invalid byte: U+0074 't'
2021/01/22 03:25:10 .../templates/helper.go:806:ActionContent2Commits() [E] json.Unmarshal:

        ERROR: unexpected end of JSON input
```

## Description
1) push to feature branch
2) hit make PR
3) 
![image](https://user-images.githubusercontent.com/8331173/105442464-d7e50800-5c37-11eb-9df8-e50e3e887072.png)

...

**Comment by zeripath** [comment]
Let me guess - the feature branch is 40 characters long

**Comment by zeripath** [comment]
This was fixed by #14624 which was merged as commit 0a23079485b8f40bc86f6869a202be8be2727cc6 e.g. v1.14.0-dev-682-g0a2307948

It would be very helpful in future to replicate on try.gitea.io then we could prove that this fixed. Please update, try again and reopen if this is not fixed.
